client/src/main/java/jp/co/soramitsu/iroha/java/Transaction.java

```diff
@@ -51,10 +53,22 @@
     this.batchMeta = BatchMeta.newBuilder(tx.getPayload().getBatch());
   }
 
+  /**
+   * An old version of transaction sign. Uses Iroha builtin Ed25519/Sha3 signature implicitly. Prefer explicit version
+   * sign(KeyPair keyPair, SignatureBuilder signatureBuilder)
+   */
+  @Deprecated
   @Override
   public BuildableAndSignable<TransactionOuterClass.Transaction> sign(KeyPair keyPair) {
     updatePayload();
-    tx.addSignatures(Utils.sign(this, keyPair));
+    tx.addSignatures(new Ed25519Sha3SignatureBuilder().sign(this, keyPair));
```

Comment:
maybe it's good idea to have the single static instance of `Ed25519Sha3SignatureBuilder` and reuse it everywhere

Reply:
added singleton, please take a look